spec/models/invite_spec.rb

```diff
@@ -317,6 +317,31 @@
       Invite.redeem_from_email('test2@example.com')
       expect(invite.reload).not_to be_redeemed
     end
+
+    it 'does not work with expired invites' do
+      invite.update!(expires_at: 1.day.ago)
+      Invite.redeem_from_email(user.email)
+      expect(invite).not_to be_redeemed
+    end
+
+    it 'does not work with deleted invites' do
+      invite.trash!
+      Invite.redeem_from_email(user.email)
+      expect(invite).not_to be_redeemed
+    end
+
+    it 'does not work with destroyed invites' do
+      invite.destroy!
+      Invite.redeem_from_email(user.email)
+      expect(invite).not_to be_redeemed
+    end
```

Comment:
I'm not sure if this test make sense. If an invite has already been removed from the database, it is impossible for it to be redeemed.

Reply:
I added this because we have a similar test for `redeem` just above:

https://github.com/discourse/discourse/blob/3eaac567978026064bbe004cfe1d3dd4cf0091a7/spec/models/invite_spec.rb#L242-L245

Reply:
I don't think the test case above make sense either. `invite.destroy!` already indicates that the row is removed from the database so `invite.redeem` essentially does nothing.

Reply:
Okay, I removed both the test cases.